fhir-profile/src/main/java/com/ibm/fhir/profile/ConstraintGenerator.java

```diff
@@ -643,7 +643,15 @@ private String getIdentifier(ElementDefinition elementDefinition) {
     private List<Type> getTypes(ElementDefinition elementDefinition) {
         if (elementDefinition.getContentReference() != null) {
             String contentReference = elementDefinition.getContentReference().getValue();
-            return elementDefinitionMap.get(contentReference.substring(1)).getType();
+            int index = contentReference.indexOf("#");
```

Comment:
indexOf('char') would probably be preferred here

Reply:
Since the constraint is compiled one time, I think it's a nominal benefit

Reply:
Per the Sonar link:
https://rules.sonarsource.com/java/RSPEC-3027?search=indexOf

It looks like this rule has been deprecated likely because modern compilers already do the optimization.